src/main/java/java/lang/StringBuffer.java

```diff
@@ -531,11 +531,10 @@ public synchronized StringBuffer append(char c) {
      */
     @Override
     public synchronized StringBuffer append(int i) {
-        // DIFFBLUE MODEL LIBRARY this is replaced internally
         // toStringCache = null;
         // super.append(i);
         // return this;
-        return CProver.nondetWithoutNullForNotModelled();
+        return append(String.valueOf(i));
```

Comment:
:pick: This isn't related to floats so should probably be in a separate commit.
:grey_question: Why is this done for int but not long, char etc.?

Reply:
Good point. It's a follow-up from https://github.com/diffblue/cbmc/pull/4633. I've split this into a separate commit.

Other types should be coming later too, but at the moment I concentrated on the bugs that I've seen in our benchmarks, and am handling them one by one.